src/headers.js

```diff
@@ -54,17 +54,24 @@ class Headers {
   append (name, value) {
     const [normalizedName, normalizedValue] = normalizeAndValidateHeaderValue(name, value)
 
-    let index = this[kHeaders].length
-    for (let i = 0; i < this[kHeaders].length; i += 2) {
-      if (normalizedName === this[kHeaders][i]) {
-        this[kHeaders][i + 1] += `, ${normalizedValue}`
-        return
-      } else if (this[kHeaders][i] > normalizedName) {
-        index = i
-        break
+    let i = this[kHeaders].length
+    let low = -1
+    let probe
+    while (i - low > 1) {
```

Comment:
can you add some comments explaining how this binary search implementation works? or if its based on some external post linking to that is sufficient